Lib/ctypes/test/test_callbacks.py

```diff
@@ -147,6 +148,20 @@ def __del__(self):
                 gc.collect()
         CFUNCTYPE(None)(lambda x=Nasty(): None)
 
+    @need_symbol('WINFUNCTYPE')
+    def test_i38748_stackCorruption(self):
+        callback_funcType = WINFUNCTYPE(c_long, c_long, c_longlong)
+        @callback_funcType
+        def callback(a, b):
+            c = a + b
+            print(f"a={a}, b={b}, c={c}")
+            return c
+        dll = cdll[_ctypes_test.__file__]
+        # With no fix for i38748, the next line will raise OSError and cause the test to fail.
+        self.assertEqual(dll._test_i38748_runCallback(callback, 5, 10), 15)
+    # Mark the above test as an expected failure on 32 bit (x86)
```

Comment:
```suggestion

    # This test is expected to fail on 32-bit (x86) because it uses stdcall
    # convention, while 64-bit will ignore that and use a safe stack layout.
```